controllers/shipwrightbuild_controller.go

```diff
@@ -17,93 +16,178 @@ import (
 	"k8s.io/apimachinery/pkg/api/errors"
 	"k8s.io/apimachinery/pkg/runtime"
 	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
 
 	"github.com/shipwright-io/operator/api/v1alpha1"
 )
 
+const (
+	// FinalizerAnnotation annotation string appended on finalizer slice.
+	FinalizerAnnotation = "finalizer.operator.shipwright.io"
```

Comment:
My main question is why do we need a finalizer? Finalizers make it difficult to tear down a Kubernetes cluster because it will block a namespace from getting deleted.

Is the concern that an admin will delete the target namespace by accident, and we'll get caught in this weird place where the operator won't work?

Reply:
I have been researching this subject, and indeed, back in ~2018 we had issues open to deal with this situation. So, I have tested this with an updated cluster, and you can delete a namespace without getting stuck. it might have happen before, but not by design.